src/main/java/uk/gov/pay/api/model/PaymentEvents.java

```diff
@@ -23,9 +24,11 @@
 
     public static PaymentEvents createPaymentEventsResponse(JsonNode payload, String paymentLink) {
         List<PaymentEvent> events = newArrayList();
-        if(payload.get("events").isArray()) {
-            for (JsonNode event : payload.get("events")) {
-                events.add(createPaymentEvent(event, paymentLink));
+        String paymentId = payload.get("charge_id").asText();
```

Comment:
As clarified (and updated in the PR commit message to be more clear) the chargeID remains at the top level as being returned from the connector, and is not there any longer in every event structure, so there won't a NPE because its looking for chargeID at the right place now. This fact has been covered by the failing E2E test first approach.

Events are not sent back without a paymentID

Will be revisiting the unit tests